homework/list.js

```diff
@@ -1,79 +1,90 @@
 'use strict';
 
 (function() {
+  
+  var ListManager = function(){
+    this._listNoteContent = [];
+    this._wrapper = null;
+  };
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  ListManager.prototype=
+  {
+     drawList() {
+      var list = this._listNoteContent;
+      var ul = document.createElement('ul');
+      ul.id = 'note-title-list';
+      var buff = document.createDocumentFragment();
+      list.forEach(function(note, i) {
+        var li = document.createElement('li');
+        li.dataset.noteId = i;
+        li.classList.add('note-title');
+        li.textContent = note.title;
+        // Note: buff is captured, so we now have a
+        // little closure naturally.
+        buff.appendChild(li);
+      });
+      ul.appendChild(buff);
+      this._wrapper.appendChild(ul);
+    },
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+    onNoteOpen(event) {
+      if (event.target.classList.contains('note-title')) {
```

Comment:
If a function is wrapped in a condition statement with only one branch, it's better to move the `if...else` into the caller, rather than callee.